apps/cowswap-frontend/src/modules/hooksStore/dapps/AirdropHookApp/hooks/usePreviewClaimableTokens.ts

```diff
@@ -0,0 +1,113 @@
+import { useCallback } from 'react'
+
+export interface PreviewClaimableTokensParams {
+  dataBaseUrl: string
+  address: string
+}
+
+type IntervalsType = { [key: string]: string }
+
+export interface RowType {
+  index: number
+  type: string
+  amount: string
+  proof: any[]
+}
+
+type ChunkDataType = { [key: string]: RowType[] }
+
+export const errors = {
+  NO_CLAIMABLE_TOKENS: "You don't have claimable tokens",
+  ERROR_FETCHING_DATA: 'There was an error trying to load claimable tokens',
+  NO_CLAIMABLE_AIRDROPS: 'You possibly have other items to claim, but not Airdrops',
+  UNEXPECTED_WRONG_FORMAT_DATA: 'Unexpected error fetching data: wrong format data',
+}
+
+export function findIntervalKey(name: string, intervals: IntervalsType) {
+  /* function to check if a name is inside a interval
+    intervals is in the format: {
+        "name1":"name2",
+        "name3":"name4",
+        ...
+    }
+    name4 > name3 > name2 > name1
+
+    returns the interval key if the condition is checked, else undefined
+    */
+  const keys = Object.keys(intervals)
+  const numberOfKeys = keys.length
+  let currentKeyIndex = 0
+  let found = false
+  while (!found) {
+    const currentKey = keys[currentKeyIndex]
+    if (currentKey <= name && intervals[currentKey] >= name) {
+      return currentKey
+      found = true
+    }
+    // Quit at once when verifying that name will not be in the intervals
+    // Imagine searching for "Albert" in a phone list, but you've finished the "A" section
+    if (currentKey > name) {
+      return undefined
```

Comment:
The idea is that the function output will be undefined if the address is not in the intervals. Notice that there are many undefined returns, because this is the behavior expected outside it.